spec/routing/single_page_routing_spec.rb

```diff
@@ -0,0 +1,18 @@
+RSpec.describe 'single page endpoint routing' do
+  it 'routes /single-item/very/long/base/path' do
+    expect(get: '/single_page/very/long/base/path').to route_to(
+      controller: 'single_item',
+      action: 'show',
+      format: :json,
+      base_path: 'very/long/base/path'
+    )
+  end
+
+  it 'routes /single_page (for the homepage)' do
+    expect(get: '/single_page/').to route_to(
+      controller: 'single_item',
+      action: 'show',
+      format: :json
```

Comment:
should we add `base_path: ""`?

Reply:
The assertion checks for the exact params. the base_path param is missing in this instance.